Code/Framework/AzFramework/AzFramework/Input/Devices/Gamepad/InputDeviceGamepad.cpp

```diff
@@ -144,7 +144,7 @@ namespace AzFramework
         }
 
         // Create the platform specific or custom implementation
-        m_pimpl.reset(implementationFactory ? implementationFactory(*this) : nullptr);
+        m_pimpl = (implementationFactory != nullptr) ? implementationFactory->Create(*this) : nullptr;
```

Comment:
likely not an issue here, because this should only be called once, but I believe `reset` would actually delete any existing implementation as part of a swap.

Reply:
That would actually be good since the pimp is meant to be singleton. But there isn't any path (currently) that would make this code execute more than once.

Reply:
Actually, in this case, i think `reset` and `=` will have the same effect.
https://en.cppreference.com/w/cpp/memory/unique_ptr/operator%3D 
```
3) Effectively the same as calling reset().
```